src/hooks/useOpenLoanHandler.tsx

```diff
@@ -178,11 +195,16 @@ export const useOpenLoanHandler = () => {
   }
 
   async function createLoan() {
+    if (!ethers.utils.isAddress(cfmmPoolAddr)) {
+      notifyError('Selected pair is not a valid cfmm pool pair.')
+      return
+    }
+
```

Comment:
I see this check and the other provider checks duplicated in this file. To make this cleaner, separate this into another function. You might also want to put this in utils

Reply:
This check would save only one line though, basically the notifyError line. Is it still worth it?

Reply:
well we'll probably want to separate all these small functions in utils when refactoring